clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/config/KubernetesConfiguration.java

```diff
@@ -53,6 +56,19 @@ public KubernetesConfigurationProperties kubernetesConfigurationProperties() {
     return new KubernetesConfigurationProperties();
   }
 
+  @Bean
+  @RefreshScope
+  @ConfigurationProperties("kubernetes")
+  @ConditionalOnProperty(
```

Comment:
I'm trying to understand this conditional, shouldn't we create this bean in all cases regardless of the value of `kubernetes.custom-property-binding-enabled`? What happens if this bean is not created?

Reply:
So this property `kubernetes.custom-property-binding-enabled` is an opt-in property. By default it is set to false. So this bean is the one that should be selected when it is either set to false, or not defined. That would mean, the user is letting Springboot do the properties loading and binding.

If the bean is not created, then [this bean](https://github.com/spinnaker/clouddriver/blob/00d498ed9cad0f90a4e8a862ffae5ca49ae9fccb/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/config/KubernetesCustomBinderConfiguration.java#L43-L48) is used.

Reply:
I didn't notice the other bean definition, looks good.